source/opt/vector_dce.cpp

```diff
@@ -52,6 +53,10 @@ void VectorDCE::FindLiveComponents(Function* function,
   // components are live because of arbitrary nesting of structs.
   function->ForEachInst(
       [&work_list, this, live_components](Instruction* current_inst) {
+        if (current_inst->GetOpenCL100DebugOpcode() !=
+            OpenCLDebugInfo100InstructionsMax) {
+          return;
```

Comment:
Similar comment to #3492.  Add a predicate member to class `Instruction`.  Maybe `Instruction::IsDebug()`?

Reply:
Let me update this code after submitting #3492.

Reply:
done